packages/synchronize/lib/src/sync.dart

```diff
@@ -0,0 +1,246 @@
+import 'package:synchronize/src/action.dart';
+import 'package:synchronize/src/conflict.dart';
+import 'package:synchronize/src/journal.dart';
+import 'package:synchronize/src/journal_entry.dart';
+import 'package:synchronize/src/object.dart';
+import 'package:synchronize/src/sources.dart';
+
+/// Sync between two [SyncSources]s.
+///
+/// This implementation follows https://unterwaditzer.net/2016/sync-algorithm.html in a generic and abstract way
+/// and should work for any two kinds of sources and objects.
+Future<List<SyncConflict<T1, T2>>> sync<T1, T2>(
+  final SyncSources<T1, T2> sources,
+  final SyncJournal journal,
+  final Map<String, SyncConflictSolution> conflictSolutions, {
```

Comment:
looks like the solution is more or less optional as wherever we don't have one an empty map is passed around.
I'd rather like the parameter to be optional in the first place.

This also applies to all other methods